src/libs/tmdk/e2e-tests/basic-e2e/basic-e2e.test.ts

```diff
@@ -0,0 +1,492 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  getDefaultAwsClients as aws,
+  initDefaultAwsClients,
+} from "../../src/lib/aws-clients";
+import { Arguments } from "yargs";
+import * as deploy from "../../src/commands/deploy";
+import * as init from "../../src/commands/init";
+import * as nuke from "../../src/commands/nuke";
+import * as fs from "fs";
+import * as constants from "./basic-e2e-constants";
+import {
+  twinMakerAssumeRolePolicy,
+  twinMakerPermissionPolicy,
+  twinMakerPermissionPolicySuffix,
+} from "./basic-e2e-iam";
+import {
+  ComponentTypeSummary,
+  GetComponentTypeCommandOutput,
+} from "@aws-sdk/client-iottwinmaker";
+import { DeleteObjectCommand, PutObjectCommand } from "@aws-sdk/client-s3";
+import * as path from "path";
+import { EntitySummary } from "@aws-sdk/client-iottwinmaker/dist-types/models/models_0";
+import { delay } from "../../src/lib/utils";
+import prompts = require("prompts");
+
+// Ensure nuke always operates
+prompts.inject(["Y", "Y"]);
+
+test("basic e2e test", async () => {
+  console.log("//////      BEGIN BASIC E2E TEST     //////");
+  let argv2;
+
+  // 0. Get account info
+  console.log("Initializing AWS client.");
+  initDefaultAwsClients({ region: `${constants.region}` });
+  const accountId = (await aws().getCurrentIdentity())["accountId"];
+  const twinMakerRoleName = constants.workspaceId + "-" + accountId + "-role";
+  const twinMakerPolicyName =
+    twinMakerRoleName + twinMakerPermissionPolicySuffix;
+  const twinMakerRoleArn =
+    "arn:aws:iam::" + accountId + ":role/" + twinMakerRoleName;
+  const workspaceS3BucketName = constants.workspaceId + "-" + accountId;
+  const workspaceS3BucketArn = "arn:aws:s3:::" + workspaceS3BucketName;
+  const scene1S3Location =
+    "s3://" + workspaceS3BucketName + "/" + constants.scene1FileName;
+  const model1S3Location =
+    "s3://" + workspaceS3BucketName + "/" + constants.model1FileName;
+  const model2S3Location =
+    "s3://" + workspaceS3BucketName + "/" + constants.model2FileName;
+
+  // 1. Clean up pre-existing resources, if any
```

Comment:
to avoid accidental deletion of a workspace that had issues and would like to debug, can we instead create a new  workspace (e.g. timestamp-suffixed) each time (since we should be fully cleaning up at end anyway)

we can also include something to help fully delete a workspace as you have here (or extend nuke to do so) if we end up with cruft workspaces to cleanup from integ test failures

Reply:
Sure, will add